src/plant_species/species.repository.ts

```diff
@@ -0,0 +1,158 @@
+import { NewSpeciesDTO } from "./../shared/new-species-dto";
+import { HttpException, HttpStatus, Injectable } from "@nestjs/common";
+import { Prisma } from "@prisma/client";
+import { PrismaService } from "nestjs-prisma";
+import { UpdateSpeciesDTO } from "../shared/update-species-dto";
+
+@Injectable()
+export class SpeciesRepository {
+  constructor(private readonly prisma: PrismaService) {}
+
+  async createSpecies(newSpeciesDTO: NewSpeciesDTO) {
+    try {
+      return await this.prisma.plantSpecies.create({
+        data: {
+          name: newSpeciesDTO.name,
+          OptimalCo2: newSpeciesDTO.optimalCo2
+            ? newSpeciesDTO.optimalCo2
+            : null,
+          optimalTemperature: newSpeciesDTO.optimalTemperature
+            ? newSpeciesDTO.optimalTemperature
+            : null,
+          optimalHumidity: newSpeciesDTO.optimalHumidity
+            ? newSpeciesDTO.optimalHumidity
+            : null,
+        },
+      });
+    } catch (e) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        // The .code property can be accessed in a type-safe manner
+        if (e.code === "P2002") {
+          throw new HttpException(
+            "Species with this name already exists",
+            HttpStatus.BAD_REQUEST,
+          );
+        }
+      }
+      throw e;
+    }
+  }
+
+  async updateSpecies(updateSpeciesDTO: UpdateSpeciesDTO) {
+    try {
+      const data = {};
+
+      if (updateSpeciesDTO.updateValues.optimalCo2 !== null) {
+        data["OptimalCo2"] = updateSpeciesDTO.updateValues.optimalCo2;
+      }
+      if (updateSpeciesDTO.updateValues.optimalTemperature !== null) {
+        data["optimalTemperature"] =
+          updateSpeciesDTO.updateValues.optimalTemperature;
+      }
+      if (updateSpeciesDTO.updateValues.optimalHumidity !== null) {
+        data["optimalHumidity"] = updateSpeciesDTO.updateValues.optimalHumidity;
+      }
+      data["name"] = updateSpeciesDTO.updateValues.name;
+
+      return await this.prisma.plantSpecies.update({
+        where: { name: updateSpeciesDTO.nameToBeChanged },
+        data,
+      });
+    } catch (e) {
+      if (e instanceof Prisma.PrismaClientKnownRequestError) {
+        // The .code property can be accessed in a type-safe manner
+        if (e.code === "P2025") {
+          throw new HttpException(
+            "Species with this name does not exist",
+            HttpStatus.NOT_FOUND,
+          );
+        }
+        if (e.code === "P2002") {
+          throw new HttpException(
+            "Species with this name already exists",
+            HttpStatus.BAD_REQUEST,
+          );
+        }
+      }
+      throw e;
+    }
+  }
+
+  async getAllSpecies() {
+    const data = await this.prisma.plantSpecies.findMany({
+      include: {
+        PlantBatch: {
+          select: {
+            amount: true,
+          },
+        },
+      },
+    });
+    const transformedData = {};
```

Comment:
```suggestion
    const transformedData: Record<string, {
      OptimalCo2: number;
      optimalHumidity: number;
      optimalTemperature: number;
      totalPlants: number;
    }> = {};
```

Add a type so that typescript knows that the object should contain.